android/lib/build.gradle

```diff
@@ -16,7 +16,7 @@ android {
 }
 
 dependencies {
-  compile 'com.facebook.react:react-native:0.29.2'
-  compile "com.google.android.gms:play-services-base:9.2.0"
-  compile 'com.google.android.gms:play-services-maps:9.2.0'
+  compile 'com.facebook.react:react-native:+'
+  compile "com.google.android.gms:play-services-base:8.4.0"
+  compile 'com.google.android.gms:play-services-maps:8.4.0'
```

Comment:
We've had other problems with 8.4 recently and it is a pretty old version already.
When an app requires a version of GPS that's newer than the one that is currently installed, it should automatically update in the background, so that shouldn't be a problem. I'd rather fix the issue with 9.x instead of going back to 8.4 that's 1+ year old.
As for the react-native version, there was plenty of discussion about this in other PRs already and the tl;dr is to not use the wildcard dependency. Gradle will automatically pick the latest available version if you depend on a newer one.